src/Oscoin/Node.hs

```diff
@@ -133,14 +137,29 @@ mineBlock
        , Hashable c st
        , Ord (Hash c)
        , Serialise (Hash c)
-       , Serialise Unsealed
        , AuthTree.MerkleHash (Hash c)
        )
     => NodeT c tx st s i m (Maybe (Block c tx (Sealed c s)))
 mineBlock = do
-    Handle{hEval, hConsensus, hBlockStore} <- ask
+    Handle{hEval, hConsensus, hBlockStore, hProtocol} <- ask
     time <- currentTick
-    hoist liftIO $ Consensus.mineBlock (hoistBlockStore lift hBlockStore) hConsensus hEval time
+    res  <- hoist liftIO $ Consensus.mineBlock (hoistBlockStoreReader lift hBlockStore)
+                                               hConsensus
+                                               hEval
+                                               time
+    case res of
+      Nothing -> pure Nothing
+      Just (blk, st', receipts) -> do
+          -- NOTO(adn) Here we should dispatch the block and wait for the
```

Comment:
```suggestion
          -- NOTE(adn) Here we should dispatch the block and wait for the
```
:it: :question:

Reply:
@cloudhead was merely let the suspense to build up. You know, Italians are notoriously over-dramatic 😁 :trollface: .

Reply:
ps. It's not even Italian, that would be "nota" 😂

Reply:
Hahahahaha